web/app_options.js

```diff
@@ -68,6 +68,11 @@ const defaultOptions = {
     value: 0,
     kind: OptionKind.VIEWER + OptionKind.PREFERENCE,
   },
+  defaultZoomDelay: {
+    /** @type {number} */
+    value: 400,
```

Comment:
Given that the next release should happen fairly soon, we might not want to enable the zooming-part of the patch just before that (especially with many people now being on holiday) to have more time to find and fix any potential regressions.

However if you want to land this PR *right away*, since it contains a bunch of re-factoring of existing method signatures, I'd simply suggest that we *temporarily* disable the zooming-part until after the release like so:
```suggestion
    value: typeof PDFJSDev === "undefined" || !PDFJSDev.test("GENERIC") ? 400 : -1,
```